pyop2/base.py

```diff
@@ -1620,38 +1627,12 @@ def zero(self, subset=None):
         """Zero the data associated with this :class:`Dat`
 
         :arg subset: A :class:`Subset` of entries to zero (optional)."""
-        if hasattr(self, "_zero_parloops"):
-            loops = self._zero_parloops
+        if subset is None:
+            self.data[:] = 0
```

Comment:
There's an optimisation available here where you set `self._data[:] = 0` which fills in the halo region, and then set the `halo_valid` property (or whatever it is called) to `True`.

Reply:
Done.